Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -87,6 +87,26 @@ void RenderGrid::styleDidChange(StyleDifference diff, const RenderStyle* oldStyl
         return;
 
     const RenderStyle& newStyle = this->style();
+
+    auto hasDifferentTrackSizes = [&newStyle, &oldStyle](GridTrackSizingDirection direction) {
+        auto& newTracks = newStyle.gridTrackSizes(direction);
+        auto& oldTracks = oldStyle->gridTrackSizes(direction);
+
+        if (newTracks.size() != oldTracks.size())
```

Comment:
Doesn't `Vector::operator==` handle both parts of this check?

Reply:
certainly does!
```
bool operator==(const Vector<T, inlineCapacityA, OverflowHandlerA, minCapacityA, MallocA>& a, const Vector<T, inlineCapacityB, OverflowHandlerB, minCapacityB, MallocB>& b)
{
    if (a.size() != b.size())
        return false;

    return VectorTypeOperations<T>::compare(a.data(), b.data(), a.size());
}

```

Reply:
aha thank you!